GCEm/__init__.py

```diff
@@ -7,9 +7,201 @@
+    kernel_op : {'add', 'mul'}
+        The operation to perform in order to combine the specified `kernel`s. Only used if `kernel` is a list of strings.
+    noise_variance: float
+        The noise variance to initialise the GP regression model
+    active_dims: list of int or slice or None
+        The dimensions to train the GP over (by default all of the dimensions are used)
+    name: str
+        An optional name for the emulator
+    gpu: int
+        The GPU to use (only applicable for multi-GPU) machines
+
+    Returns
+    -------
+    Emulator
+        A GCEm emulator object which can be trained and sampled from
+
+    """
+
+    from .model_adaptor import GPFlowModel
+    import tensorflow as tf
+    import gpflow
+    from gpflow.config import default_float
+
+    n_params = training_params.shape[1]
+
+    data_processors = data_processors if data_processors is not None else []
+    data_processors.extend([Recast(default_float()), Flatten()])
+    data = CubeWrapper(training_data, data_processors=data_processors)
+
+    if isinstance(kernel, list):
+        kernel = _get_gpflow_kernel(kernel, n_params, active_dims=active_dims, operator=kernel_op)
+    elif kernel is None:
+        # TODO Maybe this should just use the above mechanism?
+        kernel = gpflow.kernels.RBF(lengthscales=[0.5] * n_params, variance=0.01, active_dims=active_dims) + \
+                 gpflow.kernels.Linear(variance=[1.] * n_params, active_dims=active_dims) + \
+                 gpflow.kernels.Polynomial(variance=[1.] * n_params, active_dims=active_dims) + \
+                 gpflow.kernels.Bias(active_dims=active_dims)
+        print("WARNING: Using default kernel - be sure you understand the assumptions this implies")
+    else:
+        pass  # Use the user specified kernel
+
+    model = GPFlowModel(gpflow.models.GPR(data=(training_params, data.data_wrapper.data),
+                                          kernel=kernel,
+                                          noise_variance=tf.constant(noise_variance,
+                                          dtype=data.dtype)))
+
+    return Emulator(model, training_params, data, name=name, gpu=gpu)
+
+
+def _get_gpflow_kernel(names, n_params, active_dims=None, operator='add'):
+    """
+        Helper function for creating a single GPFlow kernel from a combination of kernel names.
+
+    Parameters
+    ----------
+    names: list of str
+        A list of the names of the kernels to be combined
+    n_params: int
+        The number of training parameters
+    active_dims: list of int or slice or None
+        The active dimensions to allow the kernel to fit
+    operator: {'add', 'mul'}
+        The operator to use to combine the kernels
+
+    Returns
+    -------
+    gpflow.kernels.Kernel
+        The combined GPFlow kernel
+    """
+    import gpflow.kernels
+    from operator import add, mul
+    from functools import reduce
+
+    kernel_dict = {
```

Comment:
I think it would be good to include the `White` kernel here as well. For emulating "noisey" data (like with the CRM, initially) the White Kernel was really important for preventing overfitting